### PR TITLE
Revew log messages and improve network resiliency

### PR DESCRIPTION
**Pull request by lukeapage**
I've reviewed all logs - Searching for occurrences of our errors/warns and evaluating whether they should be a error/warning on the basis that error/warns should be indications that something is gone wrong that is abnormal. If a network error occurs or a fairly normal event that isn't a problem occurs, including the effect of a single open api server going down and subscriptions failing over then it should not be a warn/error.

### CODE REVIEW

File: src/openapi/transport/auth.js [hunk 1]
```diff
@@ -185,12 +192,19 @@ TransportAuth.prototype.incrementErrorCounter = function(url) {
  * @param {string} url - The url/endpoint for which error count is returned.
```
Comment (yugandhar02) [inline src/openapi/transport/auth.js:192]: update or remove comment ?

File: src/openapi/transport/auth.js [hunk 2]
```diff
@@ -185,12 +192,19 @@ TransportAuth.prototype.incrementErrorCounter = function(url) {
...
-        return this.authorizationErrorCount[url];
+TransportAuth.prototype.areUrlAuthErrorsProblematic = function(
+    url,
+    authExpiry,
+) {
+    if (this.authorizationErrors.hasOwnProperty(url)) {
+        for (let i = 0; i < this.authorizationErrors[url].length; i++) {
+            if (this.authorizationErrors[url][i].authExpiry !== authExpiry) {
+                return true;
+            }
```
Comment (yugandhar02) [inline src/openapi/transport/auth.js:203]: so we will log error if request with same url with different token failed ?
may be we should update error message as well ?

File: src/openapi/transport/auth.js [hunk 3]
```diff
@@ -155,28 +149,41 @@ TransportAuth.prototype.head = makeTransportMethod('head');
...
 
 /**
- * Run debounced cleanup of error counter map
+ * Cleanup of error counter map
  */
-TransportAuth.prototype.debounceErrorCounterCleanup = function() {
-    if (this.errorCleanupTimoutId) {
-        clearTimeout(this.errorCleanupTimoutId);
+TransportAuth.prototype.cleanupAuthErrors = function() {
+    const cleanThoseBefore = Date.now() - this.authErrorsIgnoreDuration;
```
Comment (yugandhar02) [inline src/openapi/transport/auth.js:155]: the name is confusing 
we are not actually ignoring the auth errors but keeping them for certain duration to check whether the req with same url with diff authtoken failed ?

File: src/openapi/transport/auth.js (hunk 2 above)
Comment (lukeapage) [reply src/openapi/transport/auth.js:203]: yes. updated message. its to stop one problematic service making us ddos the token request url.

File: src/openapi/transport/auth.js (hunk 3 above)
Comment (lukeapage) [reply src/openapi/transport/auth.js:155]: Tried to improve the names..

**Review by yugandhar02 — APPROVED** [review]

File: src/openapi/streaming/connection/transport/websocket-transport.js [hunk 4]
```diff
@@ -192,12 +191,16 @@ function handleSocketMessage(messageEvent) {
...
         try {
             parsedMessages = parseMessage.call(this, messageEvent.data);
-        } catch (e) {
-            handleFailure.call(this, {
-                message: `Error occurred during parsing of plain WebSocket message. Message: ${e.message}`,
-                payload: e.payload,
-                payloadSize: e.payloadSize,
-            });
+        } catch (error) {
+            log.error(
```
Comment (yugandhar02) [inline src/openapi/streaming/connection/transport/websocket-transport.js:195]: log area is missing

File: src/openapi/streaming/connection/transport/websocket-transport.js [hunk 5]
```diff
@@ -360,29 +360,29 @@ WebsocketTransport.prototype.getAuthorizePromise = function(
...
-        const options = {
-            headers: {
-                'X-Request-Id': getRequestId(),
-                Authorization: authToken,
-            },
-        };
-        const url = `${this.authorizeUrl}?contextId=${contextId}`;
+    const options = {
+        headers: {
+            'X-Request-Id': getRequestId(),
```
Comment (yugandhar02) [inline src/openapi/streaming/connection/transport/websocket-transport.js:403]: Do we only use this header for batching ?
or it also serve the purpose of tracking request ?
request id sent by openapi is different than this one ?

File: src/openapi/streaming/connection/transport/websocket-transport.js [hunk 6]
```diff
@@ -69,31 +69,38 @@ function destroySocket() {
...
+}
+
 function reconnect(isImmediate) {
     if (this.reconnectCount >= DEFAULT_RECONNECT_LIMIT) {
         this.stop();
         return;
     }
 
+    if (this.reconnectTimeout && !isImmediate) {
+        log.warn(LOG_AREA, 'Reconnecting when already reconnecting');
```
Comment (yugandhar02) [inline src/openapi/streaming/connection/transport/websocket-transport.js:88]: return early ?

File: src/openapi/streaming/connection/transport/websocket-transport.js [hunk 7]
```diff
@@ -252,6 +269,23 @@ function handleSocketClose(event) {
...
+    // if we haven't got a message recently
+    // but we found a orphan recently
+    // and got a network error subscribing
+    // and our reconnectCount is 0 so we are not currently reconnecting
+    if (this.lastMessageTime < fiveSecondsAgo &&
+        this.lastOrphanFound > fiveSecondsAgo &&
+        this.lastSubscribeNetworkError > fiveSecondsAgo &&
+        this.reconnectCount === 0) {
+
+        // reconnect immediately as no need to wait - this is the initial event
```
Comment (yugandhar02) [inline src/openapi/streaming/connection/transport/websocket-transport.js:284]: can we have debug log here that we are reconnecting due to possible network error ?

File: src/openapi/streaming/connection/transport/websocket-transport.js (hunk 5 above)
Comment (lukeapage) [reply src/openapi/streaming/connection/transport/websocket-transport.js:403]: its just for tracking whether a request is a duplicate. I haven't changed this.

File: src/openapi/streaming/connection/transport/websocket-transport.js (hunk 6 above)
Comment (lukeapage) [reply src/openapi/streaming/connection/transport/websocket-transport.js:88]: I was a bit scared, but yes I think we should.

**Pull request merged**